syncto/tests/test_authentication.py

```diff
@@ -23,6 +37,26 @@ def test_should_raise_if_client_state_header_is_missing(self):
     def test_should_return_the_client_if_everything_is_fine(self):
         self.request.headers = {AUTHORIZATION_HEADER: 'Browserid 1234',
                                 CLIENT_STATE_HEADER: '12345'}
-        with mock.patch('syncto.authentication.SyncClient') as SyncClient:
-            build_sync_client(self.request)
-            SyncClient.assert_called_with('1234', '12345')
+        with mock.patch('syncto.authentication.TokenserverClient') as TSClient:
+            TSClient.return_value.get_hawk_credentials.return_value = \
+                self.credentials
+            with mock.patch('syncto.authentication.SyncClient') as SyncClient:
+                build_sync_client(self.request)
+                TSClient.assert_called_with('1234', '12345')
+                SyncClient.assert_called_with(**self.credentials)
+
+    def test_should_cache_credentials_the_second_time(self):
+        self.request.headers = {AUTHORIZATION_HEADER: 'Browserid 1234',
+                                CLIENT_STATE_HEADER: '12345'}
+        self.request.registry.cache.flush()
+        with mock.patch('syncto.authentication.TokenserverClient') as TSClient:
+            TSClient.return_value.get_hawk_credentials.return_value = \
+                self.credentials
+            with mock.patch('syncto.authentication.SyncClient') as SyncClient:
+                # First call
+                build_sync_client(self.request)
+                SyncClient.assert_called_with(**self.credentials)
+                # Second time
+                build_sync_client(self.request)
+                TSClient.assert_called_once_with('1234', '12345')
```

Comment:
nit: At first look, I didn't see the ``_once_``. Maybe emphasize with a comment ?